### PR TITLE
upcoming: [M3-8350] - Cleanup Types for Object Storage

### PR DESCRIPTION
**Pull request by jaalah-akamai**
## Depends on
- https://github.com/linode/manager/pull/10677

## Description 📝
This tackles mostly renaming with a few improvements to types which I'll point out. Essentially I'm just laying the groundwork for some future work and trying to follow spec more closely in some areas.

## Changes  🔄
- Renaming Changes: 
    - types ending in `Request` to `Payload`. We typically use the terms interchangeably but `Payload` makes more sense since `Request` could indicate other metadata.
   - In cases where we use `Response`, I'm dropping the term.
      - `ObjectStorageBucketRequestPayload` → `CreateObjectStorageBucketPayload` 
   - In places where the payload is for a some CRUD method, I added the term in front: IE: 
      - `ObjectStorageBucketRequestPayload` → `CreateObjectStorageBucketPayload`

## Target release date 🗓️
8/5/2024

## Preview 📷
N/A

## How to test 🧪
- Ensure all tests and linting passes
- Ensure Object Storage still works as intended

### CODE REVIEW

File: packages/api-v4/src/object-storage/objects.ts [hunk 1]
```diff
@@ -33,18 +45,18 @@ export const getObjectURL = (
...
-export const getObjectACL = (
-  clusterId: string,
-  bucketName: string,
-  name: string
-) =>
+export const getObjectACL = ({
+  clusterId,
+  bucket,
+  params,
+}: GetObjectStorageACLPayload) =>
```
Comment (jaalah-akamai) [inline packages/api-v4/src/object-storage/objects.ts:52]: Deconstructing this now and using our type. Intention was to follow the spec more closely https://techdocs.akamai.com/linode-api/reference/get-object-storage-bucket-acl

File: packages/api-v4/src/object-storage/types.ts [hunk 2]
```diff
@@ -1,54 +1,51 @@
...
+  permissions: ObjectStorageKeyBucketAccessPermissions;
   cluster: string;
   region?: string; // @TODO OBJ Multicluster: Remove optional indicator when API changes get released to prod
 }
 
-export interface ScopeRequest extends Omit<Scope, 'cluster'> {
-  // @TODO OBJ Multicluster: Omit 'region' as well when API changes get released to prod
-  cluster?: string;
-  region?: string;
-}
```
Comment (jaalah-akamai) [inline packages/api-v4/src/object-storage/types.ts:32]: This wasn't used.

File: packages/api-v4/src/object-storage/types.ts [hunk 3]
```diff
@@ -137,41 +134,50 @@ export interface ObjectStorageCluster {
...
 // Gen2 endpoints ('E2', 'E3') are not supported and will return null.
-export interface ObjectStorageBucketSSLResponse {
+export interface ObjectStorageBucketSSL {
   ssl: boolean | null;
 }
 
-export interface ObjectStorageBucketAccessRequest {
-  acl?: Omit<ACLType, 'custom'>;
+export interface UpdateObjectStorageBucketAccessPayload {
+  acl?: ACLType;
```
Comment (jaalah-akamai) [inline packages/api-v4/src/object-storage/types.ts:167]: No need to omit custom from here

**Review by abailly-akamai — APPROVED** [review]
Nice cleanup, thank you! This will make things far more readable and consistent

Not noticed any regression in the feature. Approving pending all tests passing

**Review by bnussman-akamai — APPROVED** [review]

**Pull request merged**